### PR TITLE
Malfunctioning AIs get a discount on the Doomsday equipment by hacking Head of Staff APCs

### PR DESCRIPTION
**Pull request by zxaber**
## About The Pull Request
Reduces the price of the Doomsday equipment by 20 PT for each APC hacked in a Head of Staff office, as well as the Vault.
## Why It's Good For The Game
See #71404 for the prior PR and my full reasoning.

Long-story short, activating the Doomsday before having a plan in place is suicide, and it takes 13 APCs to unlock. Since there are few well hidden APCs in general, you'll usually be gathering APCs after going loud (such as with a borg machine). 13 APCs is 13 minutes, and by the time you've gathered your malfbux, you're either already dead or have already taken full control.

I had intended to give Doomsday a flat 70 PT price, but concerns were raised that an AI could conceivably hack only APCs on their sat (and perhaps one on the Lavaland outpost) and Doomsday without ever really touching the station*. So a compromise was proposed, we instead give the AI discounts by hacking Head of Staff areas, and the Vault, which are usually situated in well-traveled areas, and also have some fluff reasoning.

*I still think it'd be suicide to do this, but it's not a hill I want to die on.
## Changelog
:cl:
balance: Malf AIs that hack Head of Staff and Vault APCs will now find a discount issued on Doomsday.
/:cl:

### CODE REVIEW

**Comment by Singul0** [comment]
This is good, especially after they made malf AI only happens during midrounds in manny.

**Comment by Vect0r2** [comment]
Fuck I am bad at this. All I was gonna say is the description should also say how the vault also discounts from doomsday.
Also, what if head of staff APCs just gave you more processing power?

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 1]
```diff
@@ -981,20 +981,34 @@
...
-		apc.locked = TRUE
-		apc.coverlocked = TRUE
+		return
 
-		playsound(get_turf(src), 'sound/machines/ding.ogg', 50, TRUE, ignore_walls = FALSE)
-		to_chat(src, "Hack complete. [apc] is now under your exclusive control.")
-		apc.update_appearance()
+	malf_picker.processing_time += 10
+	var/area/apcarea = apc.area
+	if(istype(apcarea, /area/station/command/heads_quarters) || istype(apcarea, /area/station/ai_monitored/command/nuke_storage))
```
Comment (Jacquerel) [inline code/modules/mob/living/silicon/ai/ai.dm:992]: I would replace this with an `is_type_in_list` and then make "`doomsday_bonus_apcs`" (or whatever you want to call the list of APCs which give you bonus points) either a global list or a static list on a relevant datum, so it's easy to mess with in the future.

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 2]
```diff
@@ -981,20 +981,34 @@
...
+		var/datum/ai_module/destructive/nuke_station/doom_n_boom = locate(/datum/ai_module/destructive/nuke_station) in malf_picker.possible_modules["Destructive Modules"]
+		if(doom_n_boom)
+			if(!(apcarea in doom_n_boom.hacked_command_areas))
```
Comment (Jacquerel) [inline code/modules/mob/living/silicon/ai/ai.dm:995]:
```suggestion
		if(doom_n_boom && !(apcarea in doom_n_boom.hacked_command_areas))
```
Would be great if you could early return here, probably can't without moving this into a proc.
On the other hand, as this isn't hot there isn't a strong reason you _couldn't_ make this into a proc.

**Review by Jacquerel — CHANGES_REQUESTED** [review]
Couple of little changes

**Comment by YakumoChen** [comment]
8 heads plus the vault...

Does that mean the power becomes literally free after you hack most of these APCs?

**Comment by zxaber** [comment]
There's a minimum price of 50 PTs.

**Comment by JohnFulpWillard** [comment]
The vault is generally hidden compared to head of staff offices, are you sure we should include it?

**Comment by zxaber** [comment]
Honestly, as a Malf AI it's not something I would consider safe from discovery. People enter the vault semi-regularly to grab gold, steal credits, etc.

File: code/modules/mob/living/silicon/ai/ai.dm (hunk 1 above)
Comment (zxaber) [reply code/modules/mob/living/silicon/ai/ai.dm:992]: Added under the `/datum/ai_module/destructive/nuke_station` object as `var/list/discount_areas`.

**Pull request merged**